test/gurka/test_build_admin.cc

```diff
@@ -0,0 +1,245 @@
+    EXPECT_TRUE(created);
+  }
+
+  valhalla::gurka::map admin_map = BuildPBF(workdir);
+
+  boost::property_tree::ptree& pt = admin_map.config;
+  pt.put("mjolnir.concurrency", 1);
+  pt.put("mjolnir.id_table_size", 1000);
+  pt.put("mjolnir.tile_dir", workdir + "/tiles");
+  pt.put("mjolnir.admin", workdir + "/admin.sqlite");
+  pt.put("mjolnir.timezone", workdir + "/not_needed.sqlite");
+
+  std::string dbname = pt.get<std::string>("mjolnir.admin");
+
+  // Given map.pbf, BuildAdminFromPBF() creates test/data/admin.sqlite.
+  std::vector<std::string> input_files = {workdir + "/map.pbf"};
+  BuildAdminFromPBF(pt.get_child("mjolnir"), input_files);
+
+  // Load the sqlite and read the countries/states from the admin table
+  std::set<std::string> countries, states;
+  GetAdminData(dbname, countries, states);
+
+  std::set<std::string> exp_countries = {"USA"};
+  EXPECT_EQ(countries, exp_countries);
+
+  std::set<std::string> exp_states = {"Colorado", "Utah"};
+  EXPECT_EQ(states, exp_states);
+
+  //======================================================================
+  // part II: build the tile data. Call GetAdminInfo() which reads
+  // sqlite and tile data. Confirm the polygonal data is correct
+  // with a few simple containment tests.
+  //======================================================================
+  BuildTiles(admin_map, input_files);
+
+  // reverse engineer the tile_id from the nodes that make up
+  // our mock map.pbf. Its probably overkill to check every node
+  // for its tile-id if I know they are all in the same tile-id,
+  // but its a good exercise.
+  std::unordered_set<GraphId> tile_ids;
+  for (const auto& node : admin_map.nodes) {
+    const midgard::PointLL& latlon = node.second;
+    GraphId tile_id = TileHierarchy::GetGraphId(latlon, 0);
+    tile_ids.insert(tile_id);
+  }
+  EXPECT_EQ(tile_ids.size(), 1);
+
+  GraphId tile_id(*tile_ids.begin());
+
+  // Create a GraphReader so we can create a GraphTileBuilder
+  GraphReader graph_reader(pt.get_child("mjolnir"));
+  auto t = GraphTile::Create(graph_reader.tile_dir(), tile_id);
+  EXPECT_TRUE(t);
+  GraphTileBuilder tilebuilder(graph_reader.tile_dir(), tile_id, true);
+
+  // Load the admin.sqlite table we just created
+  sqlite3* db_handle = nullptr;
+  uint32_t ret = sqlite3_open_v2(dbname.c_str(), &db_handle, SQLITE_OPEN_READONLY, nullptr);
+  EXPECT_EQ(ret, SQLITE_OK);
+
+  // Call GetAdminInfo.
+  std::unordered_multimap<uint32_t, multi_polygon_type> polys;
+  std::unordered_map<uint32_t, bool> drive_on_right;
+  std::unordered_map<uint32_t, bool> allow_intersection_names;
+  AABB2<PointLL> world_box(-180.0f, -90.0f, 180.f, 90.f);
+  polys = GetAdminInfo(db_handle, drive_on_right, allow_intersection_names, world_box, tilebuilder);
+  sqlite3_close(db_handle);
+
+  // For two states part of one country, there should be three polys, etc.
+  EXPECT_EQ(polys.size(), 3);
+  EXPECT_EQ(drive_on_right.size(), 3);
+  EXPECT_EQ(allow_intersection_names.size(), 3);
```

Comment:
im not sure this is needed. below where you test the contents of the tile you can assert that everything you're testing here also happened correctly. we have other tests that exorcise tilebuilder, i would suggest adding a few bits below to cover what is covered here. 

1. you can check if its drive on right or left down below and check the actual value rather than just the existance of the property. have a look at nodeinfo::drive_on_right
2. you can check the number of polygons above where you have your sets of names above
3. the allow intersections names boolean is a bit trickier. first you have to modify your json configuration to tell which countries should have named intersections, maybe you need to add another country into the test to really see both allowed and disallowed. then you'll simply give the nodes names in your osm tags above. finally you can then check nodeinfo::named_intersection and if you really want you can even go fetch the name to make sure it matches your tag value

let me know if any of this is unclear!

Reply:
Thanks for our chat today.  I think I've got everything wrapped.  Let me know.